src/Servers/HttpSys/src/RequestProcessing/ResponseStreamAsyncResult.cs

```diff
@@ -18,6 +18,7 @@ internal sealed unsafe partial class ResponseStreamAsyncResult : IAsyncResult, I
     private uint _bytesSent;
     private readonly CancellationToken _cancellationToken;
     private readonly CancellationTokenRegistration _cancellationRegistration;
+    private UnmanagedBufferAllocator _unmanagedAllocator;
```

Comment:
Is this required for correctness to fix a pre-existing bug?

I am not able to review this for thread-safety. Can two different threads end up using the `_unmanagedAllocator` at the same time?

Also, given that this is async, it has higher chance of allocating the native memory on one thread and freeing it on a different thread. There is typically some penalty for this.

Reply:
The threading here is non-trivial. I was going to take the simpler path, just create one at the callsite, but decided to encapsulate the logic inside the async result itself. The performance considerations are interesting here. As far as correctness (i.e., is the async result only accessed from a thread at a time), that was an assumption on my part. I can remove the member here instead.